mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -114,10 +115,10 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
 
-    val distinctSplits = splits.distinct
+    val distinctSplits = splits.filter(!_.isNaN).distinct
```

Comment:
hmm, if we filter NaN before the data, with size m, goes to approxQuantile, the increased complexity would be o(m), filter NaN in the splits would normally be more cheap, with number of splits far less than that of entire dataset.